controllers/catalogController.js

```diff
@@ -115,5 +114,28 @@ module.exports = (connection) => {
     res.render("catalog", options);
   };
 
+  controllerMethods.renderPart = async (req, res) => {
+    const partId = req.params.id;
+
+    
+    let part; 
+    if (req.user) {
+      part = await catalogModel.getPartInfoByIdAndUser(
```

Comment:
Are you sure that `catalogModel` should know about current user?

Reply:
I should know if this part in basket or no

Reply:
`catalog` is not the same as `basket`. Catalog information should not depend on user.